components/AddSpot/index.js

```diff
@@ -0,0 +1,230 @@
+    return mockData.some(
+      (spot) => spot.name.toLowerCase() === spotName.toLowerCase()
+    );
+  };
+
+  return (
+    <AddSpotForm onSubmit={handleFormSubmit}>
+      <SpotName
+        id="spotName"
+        name="spotName"
+        type="input"
+        placeholder={
+          isFocused || spotName !== "NAME THIS SPOT HERE"
+            ? ""
+            : "NAME THIS SPOT HERE"
+        }
+        value={spotName}
+        onChange={handleSpotNameChange}
+        onFocus={handleFocusSpotName}
+        onBlur={handleBlurSpotName}
+        required
+        aria-label="Spotname Input"
+      />
+      <FormContainer>
+        <InputLabelWrapper>
+          <AddSpotLabel type="text" htmlFor="longitude">
+            Longitude:{" "}
+          </AddSpotLabel>
+          <AddSpotInput
+            type="input"
+            pattern="-?\d+(\.\d+)?"
+            value={longitude}
+            onChange={handleLongitudeChange}
+            required
+            id="longitude"
+            name="longitude"
+            aria-label="Longitude Input"
+          />
+        </InputLabelWrapper>
+        <InputLabelWrapper>
+          <AddSpotLabel type="text" htmlFor="latitude">
+            Latitude:{" "}
+          </AddSpotLabel>
+          <AddSpotInput
+            type="input"
+            pattern="-?\d+(\.\d+)?"
+            value={latitude}
+            onChange={handleLatitudeChange}
+            required
+            id="latitude"
+            name="latidude"
+            aria-label="Latidude Input"
+          />
+        </InputLabelWrapper>
+        <SubmitButton
+          type="submit"
+          name="create-spot"
+          aria-label="Create this spot button"
+        >
+          create this spot
+        </SubmitButton>
+        {isNameError && <Error>Please enter a spot name.</Error>}
+        {isDuplicateNameError && (
+          <Error>This spot name is already taken.</Error>
+        )}
+      </FormContainer>
+    </AddSpotForm>
+  );
+}
+
+const AddSpotForm = styled.form`
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+`;
+
+const SpotName = styled.input`
+  width: 100%;
+  font-weight: 500;
+  text-align: center;
+  text-transform: uppercase;
+  font-size: 1rem;
+  list-style: none;
+  padding: 5px 0px;
+  margin-bottom: 40px;
+  background-color: #fcfcfc;
+  border-bottom: solid 1px #d5d5d5;
+  border-top: solid 5px #2f6673;
+  border-left: 0px;
+  border-right: 0px;
+
+  &:focus {
+    outline: none;
+    background-color: #ffffff;
+  }
+`;
+
+const FormContainer = styled.div`
+  width: 250px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const InputLabelWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  margin-bottom: 20px;
+`;
+
+const AddSpotLabel = styled.label`
+  display: block;
+  font-size: 1.2rem;
+  color: #2f6673;
+  list-style-type: none;
+  margin-right: 10px;
+`;
```

Comment:
Styles like these you could export in a styles.js

Reply:
@HannesOster, thank you for your approval and your feedback. i have adjusted the necessary things.